catalog/resource_sql_table.go

```diff
@@ -57,6 +57,10 @@ func (ti *SqlTableInfo) FullName() string {
 	return fmt.Sprintf("%s.%s.%s", ti.CatalogName, ti.SchemaName, ti.Name)
 }
 
+func parseComment(s string) string {
+	return strings.ReplaceAll(s, "'", `\'`)
```

Comment:
what if we already have it escaped by `\`?

Reply:
To not break comments that have single quotes escaped by `\`,, the following change in commit 6bbe0b12d8489b3c409bac270a553b9255c7bf4b was made:
```go
func parseComment(s string) string {
	return strings.ReplaceAll(strings.ReplaceAll(s, `\'`, `'`), `'`, `\'`)
}
```

Right now the user can make sure that the single quotation mark will be escaped in the SQL query in the following ways in Terraform:
```terraform
  comment = <<-EOT
    single quote in the comment \'
  EOT
```
or 
```terraform
comment = "single quote in the comment \\'"
```
However, the comment will be flagged as an update, next time Terraform script is executed, as the string read from the table will not include the escape backslash:
```terrafrom
  ~ resource "databricks_sql_table" "abc" {
      ~ comment            = <<-EOT
          - single quote in the comment '
          + single quote in the comment \'
        EOT
```

The proposed change will not change the behavior for users who already escape with `\\'` (in Terrafrom double-quoted string) or with `\'` in Terrafrom heredoc string. For new table or column comments, users will no longer be required to escape the single quotation and comments will not be falsely flagged as updates for that reason.

Reply:
To not break comments that have single quotes escaped by `\`,, the following change in commit 6bbe0b12d8489b3c409bac270a553b9255c7bf4b was made:
```go
func parseComment(s string) string {
	return strings.ReplaceAll(strings.ReplaceAll(s, `\'`, `'`), `'`, `\'`)
}
```

Right now the user can make sure that the single quotation mark will be escaped in the SQL query in the following ways in Terraform:
```terraform
  comment = <<-EOT
    single quote in the comment \'
  EOT
```
or 
```terraform
comment = "single quote in the comment \\'"
```
However, the comment will be flagged as an update, next time Terraform script is executed, as the string read from the table will not include the escape backslash:
```terrafrom
  ~ resource "databricks_sql_table" "abc" {
      ~ comment            = <<-EOT
          - single quote in the comment '
          + single quote in the comment \'
        EOT
```

The proposed change will not change the behavior for users who already escape with `\\'` (in Terrafrom double-quoted string) or with `\'` in Terrafrom heredoc string. For new table or column comments, users will no longer be required to escape the single quotation and comments will not be falsely flagged as updates for that reason.